Firestore/core/test/firebase/firestore/util/status_test.cc

```diff
@@ -55,13 +55,46 @@ TEST(Status, Copy) {
   ASSERT_EQ(a.ToString(), b.ToString());
 }
 
+TEST(Status, Move) {
+  Status s(Status(Error::InvalidArgument, "Invalid"));
+  ASSERT_EQ(Error::InvalidArgument, s.code());
+
+  Status new_s = std::move(s);
+  ASSERT_EQ(Error::InvalidArgument, new_s.code());
+
+  Status ok = Status::OK();
+  Status new_ok = std::move(ok);
+  ASSERT_TRUE(new_ok.ok());
+  // Moved OK is not OK anymore.
+  ASSERT_FALSE(ok.ok());
+}
+
 TEST(Status, Assign) {
   Status a(Error::InvalidArgument, "Invalid");
   Status b;
   b = a;
   ASSERT_EQ(a.ToString(), b.ToString());
 }
 
+TEST(Status, MoveAssign) {
+  Status ok;
+  Status reassigned{Error::InvalidArgument, "Foo"};
+  reassigned = std::move(ok);
+  ASSERT_EQ(reassigned, Status::OK());
```

Comment:
Adding

```c++
  ASSERT_EQ(ok, Status::OK());
```
right after the line above results in a crash. That doesn't seem right.

Testing `s.ok()` should be equivalent to `s == Status::OK()`, right?

Reply:
Added assert before accessing pointer, this would still crash, but with better message.